#!/usr/bin/env bash
#SBATCH --job-name=astral
#SBATCH -c 1
#SBATCH --time 01:10:00
#SBATCH --mem-per-cpu 1G
#SBATCH -o /home/crq857/projects/Chapter2/slurmout_Oct/astral_extract_%A_%a.out
#SBATCH -e /home/crq857/projects/Chapter2/slurmout_Oct/astral_extract_%A_%a.err
#SBATCH --array=10-5000%10

BEDFILE=$(sed "${SLURM_ARRAY_TASK_ID}q;d" /projects/mjolnir1/people/crq857/Chapter2/05_Phylogenomics/bedfiles/listfiles.txt)
DIR=/projects/mjolnir1/people/crq857/Chapter2/05_Phylogenomics/input
VCF=/projects/mjolnir1/people/crq857/Chapter2/03_GeneflowAnalyses/Treemix/files/Autosomes_filtered_noindels_noastrick_diploid_minQ30_biallelic_maxmiss0.9_noadmixwolf_treemix.recode.vcf

module load bedtools

bedtools intersect -a ${VCF} -b /projects/mjolnir1/people/crq857/Chapter2/05_Phylogenomics/bedfiles/${BEDFILE} > ${DIR}/Autosomes_noadmixwolves_${BEDFILE}.vcf

#This script extracts the random start and end positions generated by the previous script from your autosomal vcf file. 
